cmd/zpool/zpool_main.c

```diff
@@ -2209,34 +2209,29 @@ do_import(nvlist_t *config, const char *newname, const char *mntopts,
 		(void) fprintf(stderr, gettext("cannot import '%s': pool "
 		    "is formatted using an unsupported ZFS version\n"), name);
 		return (1);
-	} else if (state != POOL_STATE_EXPORTED &&
+	} else if (zfs_forceimport_required(config) && 
 	    !(flags & ZFS_IMPORT_ANY_HOST)) {
 		uint64_t hostid = 0;
-		unsigned long system_hostid = get_system_hostid();
+		char *hostname;
+		uint64_t timestamp;
+		time_t t;
 
 		(void) nvlist_lookup_uint64(config, ZPOOL_CONFIG_HOSTID,
 		    &hostid);
```

Comment:
The local `t` was to make sure the right type was passed to `localtime()` to avoid an ugly cast.

We can leave it as a separate printf if you prefer, I'm not sure why it was split like this originally.  Now that you mention it this might have been original written this way to aid with translation.